platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/DefaultTestEventReporterFactory.java

```diff
@@ -47,10 +64,27 @@ public GroupTestEventReporter createTestEventReporter(String rootName) {
 
         // Renders console output for the task
         testListenerInternalBroadcaster.add(new SimpleTestEventLogger(textOutputFactory));
+
         // Emits progress logger events
         testListenerInternalBroadcaster.add(new TestEventProgressListener(progressLoggerFactory));
 
-        IdGenerator<?> idGenerator = new LongIdGenerator();
-        return new LifecycleTrackingGroupTestEventReporter(new DefaultRootTestEventReporter(testListenerInternalBroadcaster.getSource(), idGenerator, new DefaultTestSuiteDescriptor(idGenerator.generateId(), rootName)));
+        // Record all emitted results to disk
+        Path resultsDir = projectLayout.getBuildDirectory().get().dir("test-results").dir(rootName).getAsFile().toPath();
```

Comment:
On second thought, let's just add these as parameters to `createTestEventReporter` for version 0 of this. That keeps the paths more explicit for now.  We'll need one for the binary results and the final report. 

We can keep the factory at Build-scope too.

Reply:
This makes using the API quite annoying. 

Especially for those who need to update all the integ tests to pick paths for the new parameters.

I don't really think people case where these files end up